File: app/src/main/java/id/poncoe/latihanmasterandroidponcoe/Java/recyclerview/ListKucingAdapter.java
package id.poncoe.latihanmasterandroidponcoe.Java.recyclerview;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import java.util.ArrayList;

import id.poncoe.latihanmasterandroidponcoe.R;

public class ListKucingAdapter extends RecyclerView.Adapter<ListKucingAdapter.ListViewHolder> {
    private ArrayList<Kucing> listKucing;
    Context c;

    public ListKucingAdapter(ArrayList<Kucing> list, Context c) {
        this.listKucing = list;
        this.c = c;
    }

    private OnItemClickCallback onItemClickCallback;
    public void setOnItemClickCallback(OnItemClickCallback onItemClickCallback) {
        this.onItemClickCallback = onItemClickCallback;
    }


    @NonNull
    @Override
    public ListViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_row_recyclerview, viewGroup, false);
        return new ListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ListViewHolder holder, int position) {
        Kucing kucing = listKucing.get(position);

        /** Glide digunakan untuk memuat sebuah gambar, baik yang sudah Anda siapkan di drawable maupun berada di server */
        Glide.with(holder.itemView.getContext())
                .load(kucing.getPhoto())
                .apply(new RequestOptions().override(55, 55).centerCropTransform())
                .into(holder.imgPhoto);

        final String images = kucing.getPhoto();
        final String title = kucing.getName();
        final String isi = kucing.getFrom();

        //BIND
        holder.imgPhoto.setImageURI(Uri.parse(images));
        holder.tvName.setText(title);
        holder.tvFrom.setText(isi);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openDetailActivity(images,title,isi);
                //onItemClickCallback.onItemClicked(listKucing.get(holder.getAdapterPosition()));
            }
        });

    }

    public interface OnItemClickCallback {
        void onItemClicked(Kucing data);
    }

    @Override
    public int getItemCount() {
        return listKucing.size();
    }

    class ListViewHolder extends RecyclerView.ViewHolder {
        ImageView imgPhoto;
        TextView tvName, tvFrom;
        ListViewHolder(View itemView) {
            super(itemView);
            imgPhoto = itemView.findViewById(R.id.img_item_photo);
            tvName = itemView.findViewById(R.id.tv_item_name);
            tvFrom = itemView.findViewById(R.id.tv_item_from);
        }
    }

    ////open activity
    private void openDetailActivity(String... details) {
        Intent i = new Intent(c, DetailM.class);
        i.putExtra("IMAGES_KEY", details[0]);
        i.putExtra("TITLE_KEY", details[1]);
        i.putExtra("ISI_KEY", details[2]);
        c.startActivity(i);

    }
}

File: app/src/main/java/id/poncoe/latihanmasterandroidponcoe/Kotlin/recyclerview/ListKucingAdapter.kt
package id.poncoe.latihanmasterandroidponcoe.Kotlin.recyclerview

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import id.poncoe.latihanmasterandroidponcoe.R

class ListKucingAdapter(val listKucing: ArrayList<Kucing>, private val context: Context) : RecyclerView.Adapter<ListKucingAdapter.ListViewHolder>() {

    private lateinit var onItemClickCallback: OnItemClickCallbackKt

    fun setOnItemClickCallback(onItemClickCallback: OnItemClickCallbackKt) {
        this.onItemClickCallback = onItemClickCallback
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ListViewHolder {
        val view: View = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_row_recyclerview, viewGroup, false)
        return ListViewHolder(view)
    }

    override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
        val (name, from, photo) = listKucing[position]

        /** Glide digunakan untuk memuat sebuah gambar, baik yang sudah Anda siapkan di drawable maupun berada di server */

        Glide.with(holder.itemView.context)
                .load(photo)
                .apply(RequestOptions().override(55, 55).centerCrop())
                .into(holder.imgPhoto)

        holder.tvName.text = name
        holder.tvFrom.text = from
        holder.itemView.setOnClickListener {
            openDetailActivity(photo,name,from)
            Toast.makeText(holder.itemView.context, "Kamu memilih " + listKucing[holder.adapterPosition].name, Toast.LENGTH_SHORT).show()
        }
    }

    interface OnItemClickCallbackKt {
        fun onItemClicked(data: Kucing)
    }

    override fun getItemCount(): Int {
        return listKucing.size
    }

    inner class ListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvName: TextView = itemView.findViewById(R.id.tv_item_name)
        var tvFrom = itemView.findViewById<TextView>(R.id.tv_item_from)
        var imgPhoto: ImageView = itemView.findViewById(R.id.img_item_photo)
    }

    private fun openDetailActivity(vararg details: String) {
        val i = Intent(context, DetailM::class.java)
        i.putExtra("IMAGES_KEY", details[0])
        i.putExtra("TITLE_KEY", details[1])
        i.putExtra("ISI_KEY", details[2])

        context.startActivity(i)

    }
}